src/main/java/eu/darkbot/api/objects/Location.java

```diff
@@ -4,42 +4,49 @@
     /**
      * Returns distance to location ox & oy
      */
-    default double distanceTo(double ox, double oy) {
-        ox -= getX();
-        oy -= getY();
-        return Math.sqrt(ox * ox + oy * oy);
-    }
+    double distanceTo(double ox, double oy);
 
     default double distanceTo(Locatable other) {
         return distanceTo(other.getX(), other.getY());
     }
 
-    default double angleTo(double ox, double oy) {
-        return Math.atan2(getY() - oy, getX() - ox);
-    }
+    /**
+     * @return angle to other location as radians.
+     */
+    double angleTo(double ox, double oy);
 
     default double angleTo(Locatable other) {
         return angleTo(other.getX(), other.getY());
     }
 
+    /**
+     * Copies current location into a new {@link Location} object and returns it.
+     */
+    Location copy();
+    Location copy(double plusX, double plusY);
+
+    default Location copy(Location other) {
+        return copy(other.getX(), other.getY());
+    }
```

Comment:
i wouldn't mix, id leave it up for someone to just:
`loc.copy().plus(other)` instead of allowing for `loc.copy(other)`, becaue it is not clear what you're copying where, are you copying loc into other? are you setting loc to be a copy of other?